addons/mail/static/src/models/user_notification_manager.js

```diff
@@ -0,0 +1,149 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr } from '@mail/model/model_field';
+
+import { url } from '@web/core/utils/urls';
+
+registerModel({
+    name: 'UserNotificationManager',
+    identifyingFields: [],
+    recordMethods: {
+        /**
+         * Send a notification, preferably a native one. If native
+         * notifications are disable or unavailable on the current
+         * platform, fallback on the notification service.
+         *
+         * @param {Object} param0
+         * @param {string} [param0.message] The body of the
+         * notification.
+         * @param {string} [param0.title] The title of the notification.
+         * @param {string} [param0.type] The type to be passed to the no
+         * service when native notifications can't be sent.
+         * @returns
+         */
+        sendNotification({ message, title, type }) {
+            if (this.canSendNativeNotification) {
+                if (!this.messaging.env.services['multi_tab'].isOnMainTab()) {
+                    return;
+                }
+                try {
+                    this._sendNativeNotification(title, message);
+                } catch (error) {
+                    // Notification without Serviceworker in Chrome Android doesn't works anymore
+                    // So we fallback to the notification service in this case
+                    // https://bugs.chromium.org/p/chromium/issues/detail?id=481856
+                    if (error.message.includes('ServiceWorkerRegistration')) {
+                        this._sendOdooNotification(message, { title, type });
+                    } else {
+                        throw error;
+                    }
+                }
+            } else {
+                this._sendOdooNotification(message, { title, type });
+            }
+        },
+        /**
+         * @private
+         * @returns {HTMLAudioElement}
+         */
+        _computeAudio() {
+            const audioElement = new Audio();
+            audioElement.src = audioElement.canPlayType("audio/ogg; codecs=vorbis")
+                ? url('/mail/static/src/audio/ting.ogg')
+                : url('mail/static/src/audio/ting.mp3');
+            return audioElement;
+        },
+        /**
+         * Determines whether or not sending native notification is
+         * allowed.
+         *
+         * @private
+         * @returns {boolean}
+         */
+        _computeCanSendNativeNotification() {
+            return (
+                this.messaging.browser.Notification &&
+                this.messaging.browser.Notification.permission === 'granted'
+            );
```

Comment:
could cast to boolean to be sure, I'm not sure what is this.messaging.browser.Notification when it's not set